onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorDynamicQuantizeMatMul.cpp

```diff
@@ -40,16 +40,21 @@ class DmlOperatorDynamicQuantizeMatMul : public DmlOperator
                 kernelCreationContext.GetTensorShapeDescription().GetOutputTensorShape(0)
             );
         }
+        MLOperatorTensorDataType ADatatype = kernelCreationContext.GetInputEdgeDescription(OnnxInputIndex::A).tensorDataType;
         MLOperatorTensorDataType BDatatype = kernelCreationContext.GetInputEdgeDescription(OnnxInputIndex::B).tensorDataType;
 
         std::vector<uint32_t> ATensorShape = kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(OnnxInputIndex::A);
+        std::vector<uint32_t> BTensorShape = kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(OnnxInputIndex::B);
         std::vector<uint32_t> ExpectedAScaleTensorShape = {1, 1, 1, 1};
         std::vector<uint32_t> ExpectedAZeroPointTensorShape = {1, 1, 1, 1};
+        auto& outputSizes = m_outputTensorDescs[0].GetSizes();
+        auto AShapeBroadcasted = std::array<uint32_t, 4> { outputSizes[0], outputSizes[1], ATensorShape.rbegin()[1], ATensorShape.rbegin()[0] };
```

Comment:
Since we're accessing field dimensions directly now, we should include a `ML_CHECK_VALID_ARGUMENT` above to verify valid accesses. e.g.

`ML_CHECK_VALID_ARGUMENT(outputSizes.size() >= 4);`